pyccel/codegen/printing/fcode.py

```diff
@@ -2213,6 +2279,36 @@ def _print_ConstructorCall(self, expr):
         code = '{0}({1})'.format(name, code_args)
         return self._get_statement(code)
 
+    def _print_NumpyUfuncBase(self, expr):
+        type_name = type(expr).__name__
+        func_name = numpy_ufunc_to_fortran[type_name]
+        code_args = ', '.join(self._print(i) for i in expr.args)
+        code = '{0}({1})'.format(func_name, code_args)
+        return self._get_statement(code)
+
+    def _print_MathFunctionBase(self, expr):
+        type_name = type(expr).__name__
+        func_name = math_function_to_fortran[type_name]
+        code_args = ', '.join(self._print(i) for i in expr.args)
+        code = '{0}({1})'.format(func_name, code_args)
+        return self._get_statement(code)
+
+    def _print_NumpySqrt(self, expr):
+        arg = expr.args[0]
+        code_args = self._print(arg)
+        if arg.dtype is NativeInteger() or arg.dtype is NativeBool():
+            code_args = 'Real({})'.format(code_args)
+        code = 'sqrt({})'.format(code_args)
```

Comment:
Why do we have the dictionary numpy_ufunc_to_fortran and then not use it? Is sqrt a special case?

Reply:
those are special cases ,sqrt need an argument of type real for example

Reply:
Ok. But we could still use the value stored in the dictionary. It just means that if we want to change the function some day (unlikely for sqrt but it has already happened for rand for example) then it is easy to know where to look to change that

Reply:
we are using the dictionary in `_print_NumpyUfuncBase`

Reply:
Yes. But the element 'sqrt' of the dictionary is never used. Either it should be used here so that changes to functions are always carried out in the same place even if it is a special case. Or it shouldn't be in the dictionary so that it is clear that this is not used

Reply:
yes it shouldn't be in the dictionary

Reply:
There are different ways to handle the special case of `numpy.sqrt`. If we stick to this approach based on strings, I also agree that `sqrt` should not be a key of the dictionary `numpy_ufunc_to_fortran`.